students/rhamersky/Lesson_10/mailroom_part6.py

```diff
@@ -0,0 +1,392 @@
+                   "\n\n"+"This donation will be put to great use!" \
+                   "\n\n"+"Your current lifetime donation is "+"${:,.02f}".format(dic_donors[donor][0]) + "."
+        str_valediction = "\n\n"+"Sincerely," \
+                          "\n\n"+"The Team"
+        thankyou_message = str_salutation+str_body+str_valediction
+        return thankyou_message
+
+############# END OF FILE CLASS #################
+
+# The Data class is where the data get manipulated.
+
+
+class Data:
+    @staticmethod
+    def update_donor(donor_name, donation_amount, initial_donation):
+        '''
+        Updates donor's donation history or creates a new donor.
+        :param donor_name: The name of the donor.
+        :param donation_amount: The amount of money the donor donated.
+        :param initial_donation: The initial donation of the donor.
+        :return:
+        '''
+        # Creates a donor name list
+        # lst_donor_name = [dic_donors[row][0] for row in range(len(dic_donors))]  # --> Added list comprehension
+        # Checks to see if donor has given before and updates history
+        dic_new_donor_names = dic_donors
+        if donor_name in dic_donors:
+            dic_new_donor_names[donor_name][1] = dic_donors[donor_name][
+                                                     1] + 1  # --> Increase amount of times donor donated
+            dic_new_donor_names[donor_name][0] = dic_donors[donor_name][
+                                                     0] + donation_amount  # --> Adds donation to history
+            lst_donor_data = dic_new_donor_names[donor_name]  # --> Creates a list of the values in the dictionary.
+            lst_donor_data.append(donation_amount)  # --> Appends donation amount to donor data list
+            dic_new_donor_names[donor_name] = lst_donor_data  # --> Replaces the old value list with lst_donor_data.
+        # Adds new donor to the donor list
+        else:
+            dic_new_donor_names[donor_name] = [donation_amount, initial_donation, donation_amount]
+        return dic_new_donor_names
+
+    @staticmethod
+    def print_donor_names():
+        '''
+        Prints donors names if the user types "list" when the program prompts the user to input a donor.
+        :return:
+        '''
+        # Creates a list just of donor names
+        print("\n" + "Donor")
+        for name in dic_donors:
+            print(name)
+
+    @staticmethod
+    def dic_sort(dic_donors):
+        '''
+        Takes a dictionary and creates a sorted tuple using the first element of the value list.
+        It then puts the sorted data into a new dictionary.
+        :param dic_donors: The dictionary of donors.
+        :return:
+        '''
+        dic_donors_total_donation = {}
+        # Removes all past donations from the value list.
+        for key in dic_donors:
+            lst_total_donation = dic_donors[key]
+            dic_donors_total_donation[key] = lst_total_donation
```

Comment:
Is this loop making an exact copy of the dict? If so, you don't need a loop for that. Also not clear on why you need a copy here